### PR TITLE
UI: Generic

### PR DESCRIPTION
**Pull request by Amstutz**
Proposal for UI Panel Components, to be discussed in the next JF. This components is used as container for legacy content.

### CODE REVIEW

File: src/UI/Component/Generic/Generic.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @package ILIAS\UI\Component\Generic
+ */
+interface Generic extends \ILIAS\UI\Component\Component {
+
+	/**
+	 * Set content as string stored in this component.
+	 * @param string $content
+	 * @return	Generic
+	 */
+	public function withContent($content);
```
Comment (klees) [inline src/UI/Component/Generic/Generic.php]: Do we really want to change the content after construction?

File: src/UI/Factory.php [hunk 2]
```diff
@@ -253,4 +253,23 @@ public function listing();
...
+	 *
+	 * rules:
+	 *   wording:
+	 *      1: This component MUST only be used to ensure backwards compatibility with existing UI elements in ILIAS.
+	 * ---
+	 *
+	 * @param   string $content
+	 * @return  \ILIAS\UI\Component\Generic\Generic
+	 */
+	public function generic($content);
```
Comment (klees) [inline src/UI/Factory.php]: Maybe something about HTML in the docu?

**Comment by klees** [comment]
Should we maybe rename that component to PlainHTML? Seems to be more clear what that component is then. Or LegacyHTML? Generic seems to be very generic to me.

**Comment by matthiaskunkel** [comment]
Jour Fixe, August 15, 2016: We prefer "Legacy" instead of "Generic". PR appreciated and can be merged.

File: src/UI/Factory.php (hunk 2 above)
Comment (Amstutz) [inline src/UI/Factory.php]: see composition -->  "The generic component contains html or any other content as string".

File: src/UI/Factory.php [hunk 3]
```diff
@@ -261,15 +261,15 @@ public function image();
 	 *     This component is used to wrap an existing ILIAS UI element into a UI component. This is useful if a container
 	 *     of the UI components needs to contain content that is not yet implement in the centralized UI components.
 	 *   composition: >
-	 *     The generic component contains html or any other content as string.
+	 *     The legacy component contains html or any other content as string.
 	 *
 	 * rules:
 	 *   wording:
 	 *      1: This component MUST only be used to ensure backwards compatibility with existing UI elements in ILIAS.
```
Comment (chfsx) [inline src/UI/Factory.php]: Maybe a sacond rule could be:
2. This component MUST only contain Elements which cannot be generated using other UI Componentf from the UI Service

Comment (Amstutz) [inline src/UI/Factory.php]: Thank you for your feedback. I personally appreciate your suggestion to stress this point further. 

Rule changes MUST be discussed by the JF. In this case however, I do not believe this to be an actual change of the rules, but a clarification of a rule already listed. I therefore commit your suggestion. However, I downgraded the wording a little to should to leave some room open for edge cases. If you prefer a more rigid wording I think we have to make a proper proposal.

Comment (chfsx) [inline src/UI/Factory.php]: I'm perfectly fine with SHOULD in this context since we can't predict all use cases like nesting

**Comment by klees** [comment]
LGMT

**Pull request merged**